src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,201 @@
+import { assert } from 'chai';
+import path = require('path');
+import {
+	ActivityBar,
+	CustomTreeSection,
+	DebugToolbar,
+	EditorView,
+	InputBox,
+	SideBarView,
+	TextEditor,
+	ViewItem,
+	VSBrowser,
+	Workbench
+} from 'vscode-extension-tester';
+import { prepareEmptyTestFolder } from './utils/resourcesUtils';
+
+const TEST_FOLDER = '../../../testFolder';
+const WORKSPACE_FOLDER = path.join(__dirname, TEST_FOLDER);
+
+const START_DEBUG_LABEL = 'Start Java debugger on Camel K integration';
+const REMOVE_INTEGRATION_LABEL = 'Remove Apache Camel K Integration';
+
+
+describe('Tooling for Apache Camel K extension', function () {
+
+	before(async function () {
+		this.timeout(200000);
+		await prepareTempWorkspaceForTests(WORKSPACE_FOLDER);
+		await VSBrowser.instance.driver.sleep(3000);
```

Comment:
sleep should be avoided as much as possible. it is introducing slowness and flakiness to test suite. it will be better to use a waitUntil with a specific condition

Reply:
I can try using waitUntil but I'm not sure what conditions do we have to know when something has finished on the ui-tests. None of the other ui-tests use it either, only the integration tests. I'll review it and see what I can do.

Reply:
My bad, Selenium is coming with its own way to provide a conditional wait using driver.wait, see for instance https://github.com/camel-tooling/vscode-camelk/blob/06972bb32366d2767c1bb2b052c5b6396ce9ee19/src/ui-test/dev_integration_test.ts#L89 (instead of waitUntil)

On what to wait, I do not know, it depends what is expected by prepareTempWorkspaceForTests, I would even say that the conditional wait shoudl be inside this method. If not possible, the driver.wait must be related to the next first action.

Reply:
I managed to change every one using the driver.wait :tada: